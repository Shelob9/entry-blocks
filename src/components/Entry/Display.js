import React, {Fragment} from 'react';
import {EntryFieldView} from "../EntryViewer/EntryFieldView";
import Grid from "react-css-grid";

export default function (props) {
	const {
		entryId,
		formId,
		entryFieldId,
		entry,
		before,
		after
	} = props;

	if( ! formId || ! entryId || ! entryFieldId ){
		return <Fragment/>
	}
	let entryField = {};
	if( entry.hasOwnProperty(entryFieldId) ){
		//id or datestamp
		entryField = {value: entry[entryFieldId]};
	}
	else if( ! entry.hasOwnProperty('fields') || ! entry.fields.hasOwnProperty(entryFieldId)){
		return <Fragment/>
	}else{
		entryField = entry.fields[entryFieldId];
	}

	return(
		<Grid>
			<Fragment>{before}</Fragment>
			<EntryFieldView entryField={entryField} fieldType={'text'} />
			<Fragment>{after}</Fragment>
		</Grid>
	);
}
