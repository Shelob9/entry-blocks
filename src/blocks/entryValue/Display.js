import React, {Fragment} from 'react';
import {EntryFieldView} from '../../components/Entry/';
import Grid from 'react-css-grid'


/**
 * Display EntryValue
 *
 * @param props
 * @return {*}
 * @constructor
 */
export default function Display(props) {
	const {
		entryId,
		formId,
		entryFieldId,
		entry,
		before,
		after
	} = props;

	if( ! formId || ! entryId || ! entryFieldId ){
		return <Fragment/>
	}

	let entryField = {};
	if( entry.fields.hasOwnProperty(entryFieldId) ){
		entryField = entry.fields[entryFieldId];
	}
	else{
		return <Fragment/>
	}

	return(
		<Grid>
			<Fragment>{before}</Fragment>
			<EntryFieldView entryField={entryField} fieldType={'text'} />
			<Fragment>{after}</Fragment>
		</Grid>
	);




};
