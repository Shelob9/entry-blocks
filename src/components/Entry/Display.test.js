import renderer from 'react-test-renderer';
import React from 'react';
import Enzyme, {shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Display from "./Display";
Enzyme.configure({adapter: new Adapter()});

describe( 'Entry display component', () => {
	const _forms = require( '../../__MOCK_DATA__/forms' );
	const forms = Object.values(_forms );
	const formId = 'CF5bde22b646ed0';
	const form = _forms[formId];
	const entries = require( '../../__MOCK_DATA__/entries' );
	const entry = entries[0];
	const entryFieldId = 'fld_8768091';
	const entryId = entry.id;
	const instanceId = '54fds';
	const beforeText = 'Before';
	const afterText = 'After';
	const onSetForm = jest.fn();
	const onSetEntry = () => jest.fn();
	const onSetEntryField =jest.fn();
	const setBefore = () => jest.fn();
	const setAfter = () => jest.fn();

	test( 'Displays when all props are provided and matches snapshot',() => {
		expect( renderer.create(
			<Display
				form={form}
				currentEntryId={entryId}
				entryId={entryId}
				entry={entry}
				formId={formId}
				entryFieldId={entryFieldId}
				before={beforeText}
				after={afterText}
			/>
		)).toMatchSnapshot();
	});


	test( 'Returns a fragment if not enough to display',() => {
		expect( renderer.create(
			<Display
				form={{}}
				currentEntryId={''}
				entryId={0}
				entry={{}}
				formId={''}
				entryFieldId={''}
				before={''}
				after={''}
			/>
		)).toMatchSnapshot();
	});

	test( 'Returns a fragment if entry has no fields',() => {
		const entry = {
			"id": "1",
			"form_id": "CF5bde22b646ed0",
			"datestamp": "2018-11-03 22:35:54",
			"status": "active",
			"user": {
				"id": "",
				"name": "admin",
				"email": ""
			},
		};
		expect( renderer.create(
			<Display
				form={form}
				currentEntryId={entryId}
				entryId={entryId}
				entry={entry}
				formId={formId}
				entryFieldId={entryFieldId}
				before={beforeText}
				after={afterText}
			/>
		)).toMatchSnapshot();
	});
});
