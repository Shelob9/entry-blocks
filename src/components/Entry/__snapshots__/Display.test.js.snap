exports[`Entry display component Displays when all props are provided and matches snapshot 1`] = `
<div
  className="sc-bdVaJa eHjUkg"
  width={320}>
  Before
  Josh
  After
</div>
`;

exports[`Entry display component Returns a fragment if entry has no fields 1`] = `null`;

exports[`Entry display component Returns a fragment if not enough to display 1`] = `null`;
